runtime/protocol/http-client-engines/test-suite/build.gradle.kts

```diff
@@ -58,67 +58,80 @@ kotlin {
     }
 }
 
-open class LocalTestServers : DefaultTask() {
-    @Internal
-    var server: Closeable? = null
-        private set
-
-    @Internal
-    lateinit var main: String
+abstract class TestServerService :
+    BuildService<TestServerService.Params>,
+    AutoCloseable {
+    interface Params : BuildServiceParameters {
+        val sslConfigPath: Property<String>
+        val mainClass: Property<String>
+        val classpath: ConfigurableFileCollection
+    }
 
-    @Internal
-    lateinit var classpath: FileCollection
+    private var server: Closeable? = null
 
-    @Input
-    lateinit var sslConfigPath: String
+    fun startServers() {
+        if (server != null) return
 
-    @TaskAction
-    fun exec() {
         try {
             println("[TestServers] start")
-            val urlClassLoaderSource = classpath.map { file -> file.toURI().toURL() }.toTypedArray()
+            val urlClassLoaderSource = parameters.classpath.map { it.toURI().toURL() }.toTypedArray()
             val loader = URLClassLoader(urlClassLoaderSource, ClassLoader.getSystemClassLoader())
 
-            val mainClass = loader.loadClass(main)
+            val mainClass = loader.loadClass(parameters.mainClass.get())
             val main = mainClass.getMethod("startServers", String::class.java)
-            server = main.invoke(null, sslConfigPath) as Closeable
+            server = main.invoke(null, parameters.sslConfigPath.get()) as Closeable
             println("[TestServers] started")
         } catch (cause: Throwable) {
             println("[TestServers] failed: ${cause.message}")
             throw cause
         }
     }
 
-    fun stop() {
-        if (server != null) {
-            server?.close()
-            println("[TestServers] stop")
-        }
+    override fun close() {
+        server?.close()
+        println("[TestServers] stopped")
+    }
+}
+
+val testServerService = gradle.sharedServices.registerIfAbsent("testServers", TestServerService::class) {
+    parameters.sslConfigPath.set(File.createTempFile("ssl-", ".cfg").absolutePath)
+    parameters.mainClass.set("aws.smithy.kotlin.runtime.http.test.util.TestServersKt")
```

Comment:
Do we / will we ever use a different value for `mainClass`? If not, we could consider hardcoding this and removing the parameter